sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2856,6 +2883,20 @@ pi_result piQueueRelease(pi_queue Queue) {
         if (ZeResult && ZeResult != ZE_RESULT_ERROR_UNINITIALIZED)
           return mapError(ZeResult);
       }
+      if (Queue->UsingImmCmdLists && Queue->OwnZeCommandQueue) {
+        std::scoped_lock<ur_mutex> Lock(
+            Queue->Context->ZeCommandListCacheMutex);
+        // Add commandlists to the cache for future use.
+        // They will be deleted when the context is destroyed.
+        pi_command_list_info_t MapEntry = it->second;
```

Comment:
avoid redundant copies:
```suggestion
        const pi_command_list_info_t &MapEntry = it->second;
```

Reply:
Done.